File: backend/src/main/java/com/campuslink/entity/Question.java
package com.campuslink.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 问题实体类
 */
@Data
@TableName("question")
public class Question {
    /**
     * 问题ID
     */
    @TableId(value = "q_id", type = IdType.AUTO)
    private Long qid;

    /**
     * 问题标题
     */
    private String title;

    /**
     * 问题内容
     */
    private String content;

    /**
     * 提问者ID
     */
    @TableField("asker_id")
    private Long askerId;

    /**
     * 问题分类 (学习/生活/技术等)
     */
    private String category;

    /**
     * 标签（JSON数组）
     */
    private String tags;

    /**
     * 图片URL列表（JSON数组）
     */
    private String images;

    /**
     * AI生成的答案
     */
    @TableField("ai_answer")
    private String aiAnswer;

    /**
     * AI答案生成时间
     */
    @TableField("ai_generated_at")
    private LocalDateTime aiGeneratedAt;

    /**
     * 浏览次数
     */
    private Integer views;

    /**
     * 回答数量
     */
    @TableField("answer_count")
    private Integer answerCount;

    /**
     * 是否已解决: 0-否, 1-是
     */
    @TableField("is_solved")
    private Integer isSolved;

    /**
     * 悬赏积分
     */
    @TableField("reward_points")
    private Integer rewardPoints;

    /**
     * 状态: 0-已删除, 1-正常
     */
    private Integer status;

    /**
     * 创建时间
     */
    @TableField("created_at")
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField("updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/question.ts
/**
 * 问答模块类型定义
 */

// 问题分类
export type QuestionCategory = '学习' | '生活' | '技术' | '其他'

// 问题列表项
export interface QuestionItem {
  qid: number              // 后端返回 qid
  questionId?: number      // 兼容字段
  title: string
  content?: string
  askerNickname: string    // 后端返回 askerNickname
  askerName?: string       // 兼容字段
  askerAvatar?: string
  askerLevel?: number      // 提问者等级
  category: QuestionCategory
  tags?: string[]
  views: number
  answerCount: number
  status: number           // 后端返回 status (0=未解决, 1=已解决)
  isSolved?: boolean       // 兼容字段
  bounty: number           // 后端返回 bounty
  rewardPoints?: number    // 兼容字段
  createdAt: string
}

// 问题详情
export interface QuestionDetail extends QuestionItem {
  askerId: number
  images?: string[]
  aiAnswer?: string
  aiGeneratedAt?: string
}

// 回答列表项
export interface AnswerItem {
  answerId: number
  questionId: number
  responderId: number
  responderName: string
  responderAvatar: string
  content: string
  images?: string[]
  likes: number
  isAccepted: boolean
  isLiked: boolean
  createdAt: string
}

// 问题列表查询参数
export interface QuestionListParams {
  category?: QuestionCategory | string | null
  isSolved?: number | null  // 0=未解决，1=已解决
  keyword?: string
  page?: number
  pageSize?: number
  sortBy?: 'views' | 'created_at' | 'bounty' | 'answerCount'
  sortOrder?: 'asc' | 'desc'
}

// 提问参数
export interface QuestionCreateParams {
  title: string
  content: string
  category: QuestionCategory
  tags: string[]
  images?: string[]
  bounty: number
}

// 回答参数
export interface AnswerCreateParams {
  content: string
  images?: string[]
}

